ideas/2024.md

```diff
@@ -114,3 +114,29 @@ medium
 Prior efforts:
 * https://discourse.nixos.org/t/nixtract-0-1-0-extract-graph-data-from-nix-code/34731
 * https://github.com/NixOS/ofborg/blob/released/ofborg/src/outpaths.nix
+
+
+### Testing and enhancing Dynamic Derivations
+
+Existing work on a experimential feature known as "Dynamic derivations," which refers to derivation outputs that can themselves be derivations, lacks the adquate testing to be upstreamed. Moreover, functinality can be added to this feature in order to enable "lang2nix" tools to act in a more seamless manner than they do now. 
+
+The goal of this project is thus to throughly test this feature, in addition to adding new functionalities to it. Such functionalities include but are not limited to:
+
+* Methods for converting foreign packages to Nix, such as with graphs for various package managers, we can also create conversion tools for Ninja and Make graphs. This could greatly benefit large builds like Chromium and LLVM.
+
+* Additionally, determine a path for upstream tools like CMake, Meson, etc., to integrate with Nix. 
+
+* We can also explore modifying language-specific package managers to utilize Nix for the actual building process.
```

Comment:
```suggestion
The goal of this project is thus to thoroughly test this feature, in addition to adding new functionalities to it. Such functionalities include but are not limited to:

* Methods for converting foreign packages to Nix, such as with graphs for various package managers, we can also create conversion tools for Ninja and Make graphs. This could greatly benefit large builds like Chromium and LLVM.

* Additionally, determine a path for upstream tools like CMake, Meson, etc., to integrate with Nix. 

* We can also explore modifying language-specific package managers to utilize Nix for the actual building process.
```